src/compile/mark/mark.ts

```diff
@@ -149,8 +149,8 @@ function detailFields(model: UnitModel): string[] {
 }
 
 function scaleClip(model: UnitModel) {
-  const xScaleDomain = model.scaleDomain(X);
-  const yScaleDomain = model.scaleDomain(Y);
-  return (xScaleDomain && isSelectionDomain(xScaleDomain)) ||
-    (yScaleDomain && isSelectionDomain(yScaleDomain)) ? true : false;
+  const xScale = model.getScaleComponent('x');
+  const yScale = model.getScaleComponent('y');
+  return (xScale && xScale.get('domainRaw')) ||
```

Comment:
Maybe add comment that we apply clipping if the interval selections is bound to scale domains,
which in turn produces `domainRaw` ? 

(It took me a while to match the commit message to this part of code.)